File: fe/src/types/stadium.ts
export interface Stadium {
  id: number;
  name: string;
  address: string;
  latitude: number;
  longitude: number;
  facilities?: string;
  hourlyRate?: number;
  availableHours?: string;
  availableDays?: string[];
  contactNumber?: string;
  imageUrls?: string;
}

// API 응답용 타입
export interface StadiumDto {
  id: number;
  name: string;
  address: string;
  latitude: number;
  longitude: number;
  teamId: number;
  teamName: string;
  facilities?: string[] | null;
  hourlyRate?: number | null;
  availableHours?: string | null;
  availableDays?: string[] | null;
  contactNumber?: string | null;
  imageUrls?: string[] | null;
  // 팀 연락처 정보  
  teamContactPhone?: string | null;
  teamKakaoId?: string | null;
}

File: be/src/main/kotlin/io/be/stadium/domain/Stadium.kt
package io.be.stadium.domain

import io.be.team.domain.Team
import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import java.time.LocalDateTime

@Entity
@Table(name = "stadiums")
data class Stadium(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,
    
    @Column(nullable = false)
    val name: String,
    
    @Column(nullable = false)
    val address: String,
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id", nullable = false)
    val team: Team,
    
    val latitude: Double,
    val longitude: Double,
    
    @Column(columnDefinition = "TEXT")
    val facilities: String? = null, // JSON 형태로 시설 정보 저장
    
    val hourlyRate: Int? = null,
    
    @Column(columnDefinition = "TEXT")
    val availableHours: String? = null, // JSON 형태로 이용 가능 시간 저장
    
    @Column(columnDefinition = "TEXT")
    val availableDays: String? = null, // JSON 배열 형태로 이용 가능 요일 저장 (MONDAY, TUESDAY, ...)
    
    val contactNumber: String? = null,
    
    @Column(columnDefinition = "TEXT")
    val imageUrls: String? = null, // JSON 배열 형태로 이미지 URLs 저장
    
    @CreationTimestamp
    val createdAt: LocalDateTime = LocalDateTime.now()
)